### DISCUSSION TITLE
Setting Frequency with RFM98PW LoRa Module

### DISCUSSION THREAD

**Discussion by tolgaeng** [jgromes/RadioLib#360] [Q&A]
Hi Jan!

We have two RFM98PW LoRa Modules and we successfully can communicate with these modules from more than 9 kilometers. But when we change our frequency from 434 to about 433.663 we cannot communicate very well. It just sucks. Using these LoRa modules with an Atmega2560 and STM32F446RET6. In Atmega, so in Radiolib we are using setFrequency function to change frequency. Is it enough to change freq with just a function. Do we need to change anything else like bandwidth? 

When we change the frequency the frequency error jumps to 32 kHz from 1 kHz in the receiver serial port.

Thanks.

**Comment by jgromes** [comment]
As far as I can tell, there should be no issues in RadioLib regarding frequency configuration - like you said, it's working correctly at 434 MHz. So yes, just calling setFrequency should be enough. Can you confirm both ends of the link are configured to the correct frequency? E.g. by transmitting from both ends and checking the transmission frequency with SDR.

It could also be an issue with hardware, e.g. incorrect frequency for your PA and matching network. However, 433.663 MHz is so close to 434 MHz that this seems unlikely.

**Reply by tolgaeng** [reply]
Thanks! As far we can see, there aren't any problem in Radiolib. We did communicate between two Atmega's with different frequencies. Now we are checking the library that we are using in STM32.
